### DISCUSSION TITLE
Next seems to move <Script> tag to the end of the body

### DISCUSSION THREAD

**Discussion by welschmoor** [vercel/next.js#55726] [Help] [answered]
### Summary

Hello,
we have a third party `<script>` embed. That script uses .parentNode to attach an `<iframe>`. But what Next does is that it seems it moves all Script tags to the end of the body which means the parent node is now the html-body. Which means the iframe is attaching itself not the specific div, but to the `<body>`, which means it is at the very end of each page, below the footer.

What are my options? Asking the company to change the script to not use a parent node and instead use class or id would be one option. Is there anything I could do to stop next from moving the script?

### Additional information

```js
<body>
 <div>
  <script></script>
 </div>
</body>
```
becomes
```
<body>
 <div></div>
 <script></script>
</body>
```
```


### Example

_No response_

**Comment by abaracedo** [comment]
You can use `strategy="beforeInteractive"` so the script will be load before anything else

**Reply by welschmoor** [reply]
Hi, [abaracedo](https://github.com/abaracedo),

Error: 
`next/script`'s `beforeInteractive` strategy should not be used outside of `pages/_document.js`. 

 No Good :D. The script is inside `<head>` this way, and additionally to that, the script does not run. Plus the above error. `lazyOnload` is the same as afterInteractive and also loads stuff at the bottom of the page.

**Reply by abaracedo** [reply, marked as answer]
That should be a Warning, not an Error. If you're using App Router you can ignore it as the warning is for Page Router, it seems the team didn't update the rules (or you can turn off that rule in your `eslint` config file).

Sorry, I misunderstood and I thought you wanted it on the `<head>` tag. I did a search in Google and I found [this](https://stackoverflow.com/questions/70685073/how-to-precisely-control-where-script-tag-is-inserted-using-next-script).

**Reply by welschmoor** [reply]
Thanks big time, will try tomorrow, even though I already think it won't work, because I have no control over the parent div of the script (it is sent to us as a stringified html, so I won't be able to insert the ref). Thanks.

**Reply by abaracedo** [reply]
Let me know if it works or it doesn't and if you found any way to solve it, no one knows if I will need to do something like that in the future :smile:

**Reply by welschmoor** [reply]
It did not work for us, because we get the Script and its parent from CMS. 
What stopped me though was mostly that I could not use useRef on a script, it does not have the ref property. I tried to use ids to query select the script but hat did not work 

Edit: it actually "works", the script is in the right place in code now, but it only moves the script, not what the script loads. So, I also had to move what the script loads to the right place (btw both must be moved, else doesn't work). So, custom logic is required for each script provider.

**Reply by diego-lipinski-de-castro** [reply]
How did you fixed this in the end?
